vcd/resource_vcd_vapp_vm.go

```diff
@@ -1041,26 +1167,32 @@ func isPrimaryNicRemoved(d *schema.ResourceData) bool {
 }
 
 func changeCpuCount(d *schema.ResourceData, vm *govcd.VM) error {
-	task, err := vm.ChangeCPUCount(d.Get("cpus").(int))
-	if err != nil {
-		return fmt.Errorf("error setting CPU count: %s", err)
-	}
-	err = task.WaitTaskCompletion()
+	vmSpecSection := vm.VM.VmSpecSection
+	description := vm.VM.Description
+	// update treats same values as changes and fails, with no values provided - no changes are made for that section
+	vmSpecSection.DiskSection = nil
+
+	vmSpecSection.NumCpus = takeIntPointer(d.Get("cpus").(int))
+	// has to come together
+	vmSpecSection.NumCoresPerSocket = takeIntPointer(d.Get("cpu_cores").(int))
+	_, err := vm.UpdateVmSpecSection(vmSpecSection, description)
 	if err != nil {
-		return err
+		return fmt.Errorf("error changing memory size: %s", err)
 	}
 	return nil
 }
 
 func changeMemorySize(d *schema.ResourceData, vm *govcd.VM) error {
-	task, err := vm.ChangeMemorySize(d.Get("memory").(int))
+	vmSpecSection := vm.VM.VmSpecSection
+	description := vm.VM.Description
+	// update treats same values as changes and fails, with no values provided - no changes are made for that section
+	vmSpecSection.DiskSection = nil
+
+	vmSpecSection.MemoryResourceMb.Configured = int64(d.Get("memory").(int))
```

Comment:
Same question as with CPU. Any reason why this does not rely on builtin govcd functions?

Reply:
same answer

Reply:
What was the answer (a link would do)?

Reply:
https://github.com/vmware/terraform-provider-vcd/pull/779#discussion_r795427812